### PR TITLE
Setup Rust Docs

### PR DESCRIPTION
**Pull request by tuddman**
Fixes https://github.com/xmtp/libxmtp/issues/455

### CODE REVIEW

File: xmtp_mls/src/identity.rs [hunk 1]
```diff
@@ -132,7 +132,7 @@ impl Identity {
         }
 
         // Register the installation with the server
-        let kp = self.new_key_package(&provider)?;
+        let kp = self.new_key_package(provider)?;
```
Comment (tuddman) [inline xmtp_mls/src/identity.rs:135]: removing this reference is not related to `rust doc` specifically; other than it no longer generates a lint warning.

File: dev/docs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+set -eou pipefail
+
+cargo doc --no-deps --workspace --document-private-items --open
```
Comment (tuddman) [inline dev/docs:4]: @neekolas Open to suggestions for what flag option(s) you want to have here when building the docs.

Comment (neekolas) [reply dev/docs:4]: Can we drop this script in favour of the one in https://github.com/xmtp/libxmtp/pull/456, which is more optimized for GH pages?

**Comment by tuddman** [comment]
Dupe of https://github.com/xmtp/libxmtp/pull/456

**Comment by neekolas** [comment]
I'll copy over your fixes for the `sql_key_store` warnings
